things/postgres/clients.go

```diff
@@ -121,3 +129,96 @@ func (repo clientRepo) RetrieveBySecret(ctx context.Context, key string) (mgclie
 
 	return mgclients.Client{}, repoerr.ErrNotFound
 }
+
+func (repo clientRepo) Delete(ctx context.Context, id string) error {
+	q := "DELETE FROM clients AS c  WHERE c.id = $1 ;"
+
+	result, err := repo.DB.ExecContext(ctx, q, id)
+	if err != nil {
+		return postgres.HandleError(repoerr.ErrRemoveEntity, err)
+	}
+	if rows, _ := result.RowsAffected(); rows == 0 {
+		return repoerr.ErrNotFound
+	}
+
+	return nil
+}
+
+func (repo clientRepo) SearchBasicInfo(ctx context.Context, pm mgclients.Page) (mgclients.ClientsPage, error) {
+	sq, tq := constructSearchQuery(pm)
+
+	q := fmt.Sprintf(`SELECT c.id, c.name, c.created_at, c.updated_at FROM clients c %s LIMIT :limit OFFSET :offset;`, sq)
+	dbPage, err := pgclients.ToDBClientsPage(pm)
+	if err != nil {
+		return mgclients.ClientsPage{}, errors.Wrap(repoerr.ErrFailedToRetrieveAllGroups, err)
+	}
+
+	rows, err := repo.DB.NamedQueryContext(ctx, q, dbPage)
+	if err != nil {
+		return mgclients.ClientsPage{}, errors.Wrap(repoerr.ErrFailedToRetrieveAllGroups, err)
+	}
+	defer rows.Close()
+
+	var items []mgclients.Client
+	for rows.Next() {
+		dbc := pgclients.DBClient{}
+		if err := rows.StructScan(&dbc); err != nil {
+			return mgclients.ClientsPage{}, errors.Wrap(repoerr.ErrViewEntity, err)
+		}
+
+		c, err := pgclients.ToClient(dbc)
+		if err != nil {
+			return mgclients.ClientsPage{}, err
+		}
+
+		items = append(items, c)
+	}
+
+	cq := fmt.Sprintf(`SELECT COUNT(*) FROM clients c %s;`, tq)
+	total, err := postgres.Total(ctx, repo.DB, cq, dbPage)
+	if err != nil {
+		return mgclients.ClientsPage{}, errors.Wrap(repoerr.ErrViewEntity, err)
+	}
+
+	page := mgclients.ClientsPage{
+		Clients: items,
+		Page: mgclients.Page{
+			Total:  total,
+			Offset: pm.Offset,
+			Limit:  pm.Limit,
+		},
+	}
+
+	return page, nil
+}
+
+func constructSearchQuery(pm mgclients.Page) (string, string) {
+	var query []string
+	var emq string
+	var tq string
+
+	if pm.Name != "" {
+		query = append(query, "name ~* :name")
+	}
+	if pm.Identity != "" {
+		query = append(query, "id ~* :identity")
```

Comment:
There is no need to use regular expresions, since we do not have any advanced query. LIKE or ILIKE should be a better fit and faster.